pkg/controller/common/reconciler.go

```diff
@@ -24,9 +24,20 @@ type BaseIngressReconciler struct {
 	Client client.Client
 }
 
+const (
+	// Namespace knative-serving-ingress hardcoded for now.
+	// The whole component knative-openshift-ingress is going to be moved into
+	// knative-serving-networking-openshift anyway, where it will be possible to statically determine the namespace to use.
+	smmrName      = "default"
+	smmrNamespace = "knative-serving-ingress"
+)
+
 func (r *BaseIngressReconciler) ReconcileIngress(ctx context.Context, ci networkingv1alpha1.IngressAccessor) error {
 	logger := logging.FromContext(ctx)
 
+	if len(ci.GetFinalizers()) == 0 || ci.GetFinalizers()[0] != "ocp-ingress" {
```

Comment:
This needs to go to the top of `reconcileDeletion`. Only that action should be gated by finalizers.